### PR TITLE
SNOW-1846319: Add Feature Flag for log streaming

### PR DESCRIPTION
**Pull request by sfc-gh-ashen**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description

- Feature Flag Addition: Introduced a new feature flag, ENABLE_SPCS_LOG_STREAMING, to control the rollout of the log streaming feature.
- New Exception: Introduced FeatureNotEnabledError to handle cases where a feature flag is disabled. When attempting to use log streaming while the feature flag is turned off, users will see the following error message:
"Streaming logs from spcs containers is disabled. Set the following field in your configuration to enable it: 'ENABLE_SPCS_LOG_STREAMING'."

### CODE REVIEW

File: src/snowflake/cli/api/exceptions.py [hunk 1]
```diff
@@ -229,3 +229,8 @@ def __init__(self, show_obj_query: str):
         super().__init__(
             f"Received multiple rows from result of SQL statement: {show_obj_query}. Usage of 'show_specific_object' may not be properly scoped."
         )
+
+
+class FeatureNotEnabledError(RuntimeError):
+    def __init__(self, feature_name: str):
+        super().__init__(f"The feature '{feature_name}' is not enabled.")
```
Comment (sfc-gh-tkommineni) [inline src/snowflake/cli/api/exceptions.py:236]: Streaming logs from spcs containers is disabled. Set the following field in your configuration to enable it 'ENABLE_SPCS_LOG_STREAMING'.

File: src/snowflake/cli/_plugins/spcs/services/commands.py [hunk 2]
```diff
@@ -37,7 +37,11 @@
 )
```
Comment (sfc-gh-tkommineni) [inline src/snowflake/cli/_plugins/spcs/services/commands.py:37]: Add release notes

**Review by sfc-gh-tkommineni — DISMISSED** [review]
LGTM!

File: src/snowflake/cli/_plugins/spcs/services/commands.py [hunk 3]
```diff
@@ -250,6 +254,11 @@ def logs(
     Retrieves local logs from a service container.
     """
     if follow:
+        if not FeatureFlag.ENABLE_SPCS_LOG_STREAMING.is_enabled():
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/_plugins/spcs/services/commands.py:257]: You could use `is_disabled`.

File: src/snowflake/cli/api/exceptions.py [hunk 4]
```diff
@@ -229,3 +229,13 @@ def __init__(self, show_obj_query: str):
         super().__init__(
             f"Received multiple rows from result of SQL statement: {show_obj_query}. Usage of 'show_specific_object' may not be properly scoped."
         )
+
+
+class FeatureNotEnabledError(RuntimeError):
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/api/exceptions.py:234]: If you want print this message in nice error frame you should inherit from ClickException.

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-jsikorski — APPROVED** [review]

**Pull request merged**